xapian-core/tests/api_sorting.cc

```diff
@@ -321,79 +321,77 @@ DEFINE_TESTCASE(sortfunctorremote1, remote) {
+    // max_possible should be max(max_possible of the previous weights, max_attained for the new weights).
+    TEST_EQUAL_DOUBLE(mymset.get_max_possible(), max(mymset.get_max_attained(), mymset.get_max_possible()));
 
     return true;
 }
 
 DEFINE_TESTCASE(replace_weights2, backend) {
-    // open the database (in this case a simple text file
-    // we prepared earlier)
 
     Xapian::Database mydb(get_database("apitest_onedoc"));
 
     Xapian::Enquire enquire(mydb);
 
-    // make a simple query, with one word in it - "word".
     enquire.set_query(Xapian::Query("word"));
 
-    // retrieve the top ten results (we only expect one)
+    // retrieve the top ten results (we only expect one).
     Xapian::MSet mymset = enquire.get_mset(0, 10);
 
-    // Checking for exception when the number of weights in the input exceeds the number of documents
+    // Checking for exception when the number of weights in the input exceeds the number of documents.
     vector<double> weights {-1, -2};
     TEST_EXCEPTION(Xapian::InvalidArgumentError, mymset.replace_weights(weights.begin(), weights.end()));
 
     return true;
 }
 
 DEFINE_TESTCASE(sort_existing_mset_by_relevance, backend) {
-    // open the database (in this case a simple text file
-    // we prepared earlier)
+
     Xapian::Database db = get_database("apitest_simpledata");
     Xapian::Enquire enquire(db);
     enquire.set_query(Xapian::Query("word"));
 
-    // retrieve the top results we expect only two
+    // retrieve the top results we expect only two.
     Xapian::MSet mymset = enquire.get_mset(0, 10);
 
     // Storing the docids of the two documents in the original order.
-    vector<Xapian::docid> docids;
-    Xapian::MSetIterator i = mymset.begin();
-    docids.push_back(*i);
-    i++;
-    docids.push_back(*i);
-    // assigning new weights and then re-sorting to reverse the order of relevance
+    vector<Xapian::docid> docids {*(mymset.begin()), *(++mymset.begin())};
+    /* assigning new weights and then re-sorting to reverse the order of relevance
+     * negative weights were chosen to avoid matching the positive weights returned by enquire.
+     */
     vector<double> weights {-2, -1};
     mymset.replace_weights(weights.begin(), weights.end());
     mymset.sort_by_relevance();
     int j = -1, k = 1;
-    // checking for the correct order of documents and their respective weights.
+    /* checking for the correct order of documents and their respective weights
+     * correct order should be 1.{docids[1], double(-1)} 2.{docids[0],double(-2)}.
+     */
     for (Xapian::MSetIterator m = mymset.begin(); m != mymset.end(); ++m, --j, --k) {
 	TEST_EQUAL_DOUBLE(m.get_weight(), j);
 	TEST_EQUAL(docids[k], *m);
     }
-    // checking whether the max_attained attribute was updated correctly.
+    // max_attained of the updated weights {double(-2), double(-1)} should be double(-1).
     TEST_EQUAL_DOUBLE(mymset.get_max_attained(), double(-1));
```

Comment:
`1.0` is a `double` constant in C++ (see this [somewhat dense documentation](http://en.cppreference.com/w/cpp/language/floating_literal)).